alibi_detect/od/pytorch/ensemble.py

```diff
@@ -18,93 +17,63 @@ def __init__(self):
         super().__init__()
 
     def transform(self, x: torch.Tensor):
-        return self._transform(x)
-
-    @abstractmethod
-    def _transform(self, x: torch.Tensor):
-        """Applies class transform to `torch.Tensor`
+        """Public transform method.
 
         Parameters
         ----------
         x
             `torch.Tensor` array to be transformed
         """
-        pass
+        raise NotImplementedError()
 
     @torch.no_grad()
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        return self.transform(x=x)
+        return self.transform(x)
 
 
 class FitMixinTorch(ABC):
     fitted = False
 
-    def __init__(self):
-        """Fit mixin
-
-        Utility class that provides fitted checks for alibi-detect objects that require to be fit before use.
-        """
-        super().__init__()
-
-    def fit(self, x: torch.Tensor) -> FitMixinTorch:
-        self.fitted = True
-        self._fit(x)
-        return self
+    def fit(self, x: torch.Tensor) -> 'FitMixinTorch':
+        """Public fit method.
 
-    @abstractmethod
-    def _fit(self, x: torch.Tensor):
-        """Fit on `x` tensor.
-
-        This method should be overidden on child classes.
+        The `_fit` method contains the implementation details and should be overridden on child classes. Once the
+        `_fit` method has fit the object the `fitted` attribute should be set to `True`.
 
         Parameters
         ----------
         x
-            Reference `torch.Tensor` for fitting object.
+            `torch.Tensor` to fit object on.
         """
-        pass
+        raise NotImplementedError()
+
+    def set_fitted(self):
```

Comment:
Set return type to `self`?

Actually, may we use the new `Self` type from `typing-extensions`?